### DISCUSSION TITLE
Exporting citation key to a CSV file (again)

### DISCUSSION THREAD

**Discussion by vardavas** [retorquere/zotero-better-bibtex#2584] [Q&A]
Dear Emiliano,

I am a long-time user of zotero-better-bibtex, and I must say it has been a fantastic tool that has significantly simplified my work.

Currently, I am using Zotero 6.0.26 on a Mac, and I have been trying to export my library to a CSV file while including a column for the citation key. I came across your response to a similar issue in 2016 (Exporting citation key to a CSV file #448), but I must admit I am having some trouble understanding the steps you provided.

Here's what I have done so far:

1. I accessed the Zotero preferences and went to the Better BibTeX tab.
2. Under the Export sub-tab, I selected the postscripts sub-sub-tab.
3. Based on my interpretation of your instructions in #448, I entered the code line: "var citekey = Zotero.BetterBibTeX.keymanager.get(item, 'on-export').citekey;" 
4. I proceeded to select my collection, right-clicked on it, and chose "Export Collection."
5. From there, I selected the CSV format and exported the file.

Unfortunately, these steps did not yield the desired result, as the citation key field is still missing in the exported CSV file.

I would greatly appreciate your assistance in figuring out what I might be doing wrong or if there is a different approach I should take to include the citation key in the exported CSV file.

Thank you so much for your time and support!

Best regards,
Raff

**Comment by retorquere** [comment]
Hi,

You'd have to modify the existing CSV.js translator. The way in #448 does not work anymore, but you have the citation key available as `item.citationKey` in the translator code. Adding `citationKey` to `exportedFields` should do the job.

**Comment by vardavas** [comment]
Fantastic!
Thank you. This worked nicely.
Raff
________________________________
Da: Emiliano Heyns ***@***.***>
Inviato: domenica 23 luglio 2023 04:32
A: retorquere/zotero-better-bibtex ***@***.***>
Cc: vardavas ***@***.***>; Author ***@***.***>
Oggetto: Re: [retorquere/zotero-better-bibtex] Exporting citation key to a CSV file (again) (Discussion #2584)


Hi,

You'd have to modify the existing CSV.js translator. The way in #448<https://github.com/retorquere/zotero-better-bibtex/issues/448> does not work anymore, but you have the citation key available as item.citationKey in the translator code. Adding citationKey to exportedFields should do the job.
